dataprofiler/profilers/profile_builder.py

```diff
@@ -1402,18 +1477,26 @@ def tqdm(l):
         # Newly introduced features (python3.8) improves the situation
         sample_ids = np.array(sample_ids)
 
-        # Create structured profile objects
-        new_cols = set()
-        for col in data.columns:
-            if col not in self._profile:
-                self._profile[col] = StructuredColProfiler(
+        # Create StructuredColProfilers upon initialization
+        # Record correlation between columns in data and index in _profile
+        new_cols = False
+        if not initialized:
+            for col_idx in range(data.shape[1]):
+                col_name = data.columns[col_idx]
+                # Pandas cols are int by default, but need to fuzzy match strs
+                if isinstance(col_name, str):
+                    col_name = col_name.lower()
+                # Record index in _profile corresponding to current col profile
+                self._col_name_to_idx[col_name].append(len(self._profile))
```

Comment:
do we need this to be calculated here? if this is not initialized is 
`self._col_name_to_idx = mapping_given`?

Reply:
Mapping given in the uninitialized case is a bunch of `{1:1, 2:2, 3:3, ...}` mappings, since its about column index to profile index, not column name to a list of profile indices.

Reply:
Implemented